docs/src/pages/components/data-grid/components/CustomPaginationGrid.js

```diff
@@ -18,8 +18,8 @@ function CustomPagination() {
     <Pagination
       className={classes.root}
       color="primary"
-      count={state.pagination.pageCount}
-      page={state.pagination.page + 1}
+      count={state.page.pageCount}
```

Comment:
why renaming pagination to page? 
I don't think page suits very well here

Reply:
I split `state.pagination` (handled by `useGridPagination`) into `state.page` (handled by `useGridPage`) and `state.pageSize` (handled by `useGridPageSize`)

But I can rename the `useGridPage` into `useGridPagination` and `state.page` into `state.pagination`

Reply:
I would keep `useGridPage` but rename the state to pagination. These 2 hooks should be in the `pagination` directory
Then those 2 pagination hooks share the same state which is fine.

Reply:
https://github.com/mui-org/material-ui-x/pull/2099#discussion_r673157407

Fine for me to use the same sub state called `pagination` :+1: